set.mm

```diff
@@ -510554,6 +510862,1211 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
+  cerl $a class ~RL $.
+
+  $( Syntax for ring localization operation. $)
+  crloc $a class RLocal $.
+
+  ${
+    $d r s x w a b t $.
+    $( Define the operation giving the equivalence relation used in the
+       localization of a ring ` r ` by a set ` s ` .  Two pairs
+       ` a = <. x , y >. ` and ` b = <. z , w >. ` are equivalent if there
+       exists ` t e. s ` such that ` t x. ( x x. w - z x. y ) = 0 ` .  This
+       corresponds to the usual comparison of fractions ` x / y ` and
+       ` z / w ` .  (Contributed by Thierry Arnoux, 28-Apr-2025.) $)
+    df-erl $a |- ~RL = ( r e. _V , s e. _V |-> [_ ( .r ` r ) / x ]_
+          [_ ( ( Base ` r ) X. s ) / w ]_
+       { <. a , b >. | ( ( a e. w /\ b e. w ) /\ E. t e. s
+         ( t x ( ( ( 1st ` a ) x ( 2nd ` b ) ) ( -g ` r ) ( ( 1st ` b )
+                                    x ( 2nd ` a ) ) ) ) = ( 0g ` r ) ) } ) $.
+  $}
+
+  ${
+    $d r s x w a b k $.
+    $( Define the operation giving the localization of a ring ` r ` by a given
+       set ` s ` .  The localized ring ` r RLocal s ` is the set of equivalence
+       classes of pairs of elements in ` r ` over the relation ` r ~RL s ` with
+       addition and multiplication defined naturally.  (Contributed by Thierry
+       Arnoux, 27-Apr-2025.) $)
+    df-rloc $a |- RLocal = ( r e. _V , s e. _V |-> [_ ( .r ` r ) / x ]_
+    [_ ( ( Base ` r ) X. s ) / w ]_ ( ( (
+    { <. ( Base ` ndx ) , w >. ,
+      <. ( +g ` ndx ) , ( a e. w , b e. w |-> <. ( ( ( 1st ` a ) x ( 2nd ` b
+        ) ) ( +g ` r ) ( ( 1st ` b ) x ( 2nd ` a ) ) )
+          , ( ( 2nd ` a ) x ( 2nd ` b ) ) >. ) >. ,
+      <. ( .r ` ndx ) , ( a e. w , b e. w |-> <. ( ( 1st ` a ) x ( 1st ` b ) )
+          , ( ( 2nd ` a ) x ( 2nd ` b ) ) >. ) >. } u. {
+     <. ( Scalar ` ndx ) , ( Scalar ` r ) >. ,
+     <. ( .s ` ndx ) , ( k e. ( Base ` ( Scalar ` r ) ) , a e. w
+        |-> <. ( k ( .s ` r ) ( 1st ` a ) ) , ( 2nd ` a ) >. ) >. ,
+     <. ( .i ` ndx ) , (/) >. } ) u. {
+     <. ( TopSet ` ndx ) , ( ( TopSet ` r ) tX ( ( TopSet ` r ) |`t s ) ) >. ,
+     <. ( le ` ndx ) , { <. a , b >. | ( ( a e. w /\ b e. w ) /\ ( ( 1st ` a )
+         x ( 2nd ` b ) ) ( le ` r ) ( ( 1st ` b ) x ( 2nd ` a ) ) ) } >. ,
+     <. ( dist ` ndx ) , ( a e. w , b e. w |-> ( ( ( 1st ` a ) x ( 2nd ` b ) )
+         ( dist ` r ) ( ( 1st ` b ) x ( 2nd ` a ) ) ) ) >. } ) /s ( r ~RL s ) )
+     ) $.
+  $}
+
+  ${
+    rlocval.1 $e |- B = ( Base ` R ) $.
+    rlocval.2 $e |- .0. = ( 0g ` R ) $.
+    rlocval.3 $e |- .x. = ( .r ` R ) $.
+    rlocval.4 $e |- .- = ( -g ` R ) $.
+    ${
+      $d .x. a b t w $.  $d .~ r s w x $.  $d R a b r s t w x $.
+      $d S a b r s t w x $.  $d W a b t $.  $d a b ph $.
+      erlval.w $e |- W = ( B X. S ) $.
+      erlval.q $e |- .~ = { <. a , b >. | ( ( a e. W /\ b e. W )
+        /\ E. t e. S ( t .x. ( ( ( 1st ` a ) .x. ( 2nd ` b ) )
+                           .- ( ( 1st ` b ) .x. ( 2nd ` a ) ) ) ) = .0. ) } $.
+      erlval.19 $e |- ( ph -> R e. V ) $.
```

Comment:
necessary? If R e/ _V, then B = ( Base ` R ) = (/), and therefore S = (/).

Reply:
Yes, it's not necessary. I removed that hypothesis, as well as other similar ones in https://github.com/metamath/set.mm/pull/4810/commits/2b2a9e540a51ed9c7becc4b6bb3e36ae76cc7501